### ISSUE TITLE
Specify BDI agents via population XML

### ISSUE THREAD

**Issue by dhixsingh** [agentsoz/bdi-abm-integration#13]
Currently, the way BDI agents are specified for EES is a bit messy. 

* First, All agents in the MATSim population file are treated as BDI agents (see [Main.java#L149](https://github.com/agentsoz/bdi-abm-integration/blob/8bb89b19206fc0cee9cccd5d62204b68c3c41a7c/examples/bushfire/src/main/java/io/github/agentsoz/ees/Main.java#L149)). 
* Then, a `<bdiagents>..</bdiagents>` tag that gives a count of BDI agents also exists in the main scenario file (see [Maldon600 example here](https://github.com/agentsoz/bdi-abm-integration/blob/8ee3d73975f68fae2833ff01ec8c64e01b93a06f/examples/bushfire/scenarios/mount-alexander-shire/maldon-600/scenario_main.xml#L132)), that seems to be used in a crude way. 
* Finally, Jill init also needs a count for the number of agents to initialise, and this is done by passing in the count again (see for instance the [Maldon600 test](https://github.com/agentsoz/bdi-abm-integration/blob/8ee3d73975f68fae2833ff01ec8c64e01b93a06f/examples/bushfire/src/test/java/io/github/agentsoz/ees/MainMaldon600Test.java#L47).

What we would like is this:

* Each agent in the MATSim population file that is to be a BDI agent should have a relevant key/value attribute saying so. Somthing like `isBDIAgent=true`. @kainagel could you point us to info on how to set this up?
* Then, only those specific agent IDs should have counterpart BDI agents. 
* Any total BDI agent count required (say for Jill init), before MATSim initialises can just read the xml and count the agents with the BDI  tags.  

One issue with doing this is that at the moment we do not have an explicit map from MATSim population IDs to Jill IDs so this will need to be setup first (raised #14 for this).

**Comment by osbornejr** [comment]
Further to this, it would be good if in the MATsim population file the attribute indicated which type of BDIagent the agent should be. Like BDIagent='Resident', BDIagent='Responder' etc. In this format, an agent that doesn't need a BDI correspondant could have attribute BDIagent='null'.

**Comment by dhixsingh** [comment]
@osbornejr, yes likely we want more than one attribute and the ability to add more as we need them. Seems we could do something like this:
```
<person id="0">
  <attributes>
    <attribute name="BDIAgentType" class="java.lang.String" >io.github.agentsoz.ees.agents.Resident</attribute>
  </attributes>
  <plan selected="yes">
    ...
  </plan>
</person>
```

**Comment by kainagel** [comment]
Added a test `AttributesInPopulationTest` which should explain this.  Let me know if you have questions.

**Comment by dhixsingh** [comment]
@kainagel, is `AttributesInPopulationTest` not pushed maybe? I can't see it yet.

**Comment by dhixsingh** [comment]
@kainagel thanks, `AttributesInPopulationTest` is there now since 845fc54442933312649711ea0d887eb296421365. Yes that's all we need, I can take it from here.

**Comment by dhixsingh** [comment]
Available since 71ae240. Example input is:

https://github.com/agentsoz/bdi-abm-integration/blob/71ae240beb83a004ea81ce5dc8ce75a25a09d7b2/examples/bushfire/scenarios/mount-alexander-shire/maldon-100-with-emergency-vehicles/scenario_matsim_plans_10.xml#L10-L23

And further below:

https://github.com/agentsoz/bdi-abm-integration/blob/71ae240beb83a004ea81ce5dc8ce75a25a09d7b2/examples/bushfire/scenarios/mount-alexander-shire/maldon-100-with-emergency-vehicles/scenario_matsim_plans_10.xml#L181-L194
